src/fetch.rs

```diff
@@ -1,40 +1,42 @@
+use crate::errors::*;
+use crate::registry::{registry_path, registry_url};
 use crate::{Dependency, Manifest};
 use env_proxy;
+use git2::Repository;
 use regex::Regex;
 use reqwest;
 use semver;
-use serde_json as json;
 use std::env;
-use std::io::Read;
-use std::path::Path;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use std::sync::Once;
 use std::time::Duration;
-
-use crate::errors::*;
-
-const REGISTRY_HOST: &str = "https://crates.io";
-
-#[derive(Deserialize)]
-struct Versions {
-    versions: Vec<CrateVersion>,
-}
+use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+use url::Url;
 
 #[derive(Deserialize)]
 struct CrateVersion {
-    #[serde(rename = "crate")]
     name: String,
-    #[serde(rename = "num")]
+    #[serde(rename = "vers")]
```

Comment:
has the API changed recently?

Reply:
Not for HTTP API, but for registry index format: https://doc.rust-lang.org/cargo/reference/registries.html#index-format